code/game/gamemodes/changeling/powers/hivemind.dm

```diff
@@ -12,27 +12,37 @@
 	var/datum/changeling/changeling=user.mind.changeling
 	changeling.changeling_speak = 1
 	to_chat(user, "<i><font color=#800080>Use say \":g message\" to communicate with the other changelings.</font></i>")
-	var/datum/action/changeling/hivemind_upload/S1 = new
+	var/datum/action/changeling/hivemind_pick/S1 = new
 	if(!changeling.has_sting(S1))
 		changeling.purchasedpowers+=S1
 		S1.Grant(user)
-	var/datum/action/changeling/hivemind_download/S2 = new
-	if(!changeling.has_sting(S2))
-		S2.Grant(user)
-		changeling.purchasedpowers+=S2
 	return
 
 // HIVE MIND UPLOAD/DOWNLOAD DNA
 GLOBAL_LIST_EMPTY(hivemind_bank)
 
-/datum/action/changeling/hivemind_upload
+/datum/action/changeling/hivemind_pick
 	name = "Hive Channel DNA"
-	desc = "Allows us to channel DNA in the airwaves to allow other changelings to absorb it. Costs 10 chemicals."
-	button_icon_state = "hivemind_channel"
+	desc = "Allows us to upload or absorb DNA in the airwaves. Does not count towards absorb objectives. Costs 10 chemicals."
+	button_icon_state = "hive_absorb"
 	chemical_cost = 10
 	dna_cost = -1
 
-/datum/action/changeling/hivemind_upload/sting_action(var/mob/user)
+/datum/action/changeling/hivemind_pick/sting_action(mob/user)
+	var/datum/changeling/changeling = user.mind.changeling
+	var/channel_pick = alert("Upload or Absorb DNA?", "Channel Select", "Upload", "Absorb")
+
+	if(channel_pick == "Upload")
+		dna_upload(user)
+	if(channel_pick == "Absorb")
+		if(changeling.using_stale_dna(user))//If our current DNA is the stalest, we gotta ditch it.
+			to_chat(user, "<span class='warning'>We have reached our capacity to store genetic information! We must transform before absorbing more.</span>")
+			return
+		dna_absorb(user)
+	else
+		return
```

Comment:
Non required return

Reply:
Wasn't sure if that was needed, since it's an input box thing. But I guess it returns either way.

Reply:
At the end of code paths in procs it won't be needed. Only exception of doing a return is if you want to override the return value. Say you have this:
```dm
/proc/thingy()
 . = TRUE
 if(thingies)
  ...
  return
 
if(failed)
 return FALSE
```
Not the best of examples :P but still